### PR TITLE
fix: reorder encode_partition_value() checks and add tests

### PR DESCRIPTION
**Pull request by ldacey**
# Description
The `isinstance(val, datetime)` check was after `isinstance(val, date)` which meant that it was never found. I added a test for each encoding type.

### CODE REVIEW

**Comment by wjones127** [comment]
```python
>>> from datetime import date, datetime
>>> isinstance(datetime(2021, 1, 1), date)
True
```
TIL 🤯

File: python/tests/test_table_read.py
Comment (wjones127) [inline python/tests/test_table_read.py:1]: Thank you for thoroughly testing this 🙌

**Comment by wjones127** [comment]
@ldacey if you run `make format` in the Python directory it should fix all formatting issues.

**Comment by ldacey** [comment]
> @ldacey if you run `make format` in the Python directory it should fix all formatting issues.

Nice - thanks. I was struggling with that because of my new neovim setup which I am still tinkering with.

**Review by roeap — APPROVED** [review]

**Pull request merged**